eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -15,14 +16,31 @@ steps:
 
   - ${{ each repo in parameters.Repos }}:
     - pwsh: |
+        Set-PsDebug -Trace 1
+        Write-Host "Patch 1 $(PatchFilesLocation)"
         git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
         $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
-
-        if (!(Test-Path $repoPath)) { mkdir $repoPath }
-        Remove-Item -v -r $repoPath
-        Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
-        Get-ChildItem -r $repoPath
-
+        if ('$(PatchFilesLocation)')
+        {
+          pushd ${{ repo }}
+          foreach ($file in (Get-ChildItem $(PatchFilesLocation)))
+          {
+            $result = git -c user.name="azure-sdk" -c user.email="azuresdk@microsoft.com" am $file.FullName
```

Comment:
What happens if there is not this patch doesn't apply? We don't want it to fail in that case.

Reply:
I updated it to not exit but set HasChanges to $false